pkg/auth/server/headers/headers.go

```diff
@@ -0,0 +1,28 @@
+package headers
+
+import (
+	"net/http"
+)
+
+// We cannot set HSTS by default, it has too many drawbacks in environments
+// that use self-signed certs
+const standardHeaders map[string]string{
+	// Turn off caching, it never makes sense for authorization pages
+	"Cache-Control": "no-cache, no-store",
+	"Pragma": "no-cache",
+	"Expires": "0",
+	// Use a reasonably strict Referer policy by default
+	"Referrer-Policy": "strict-origin-when-cross-origin",
+	// Do not allow embedding as that can lead to clickjacking attacks
+	"X-Frame-Options": "DENY",
+	// Add other basic scurity hygiene headers
+	"X-Content-Type-Options": "nosniff",
+	"X-DNS-Prefetch-Control": "off",
+	"X-XSS-Protection": "1; mode=block",
+}
```

Comment:
What about CSP headers, https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP ?

We might want to set script src and style src.

Reply:
Just like for HSTS I am not sure we can set a static CSP because branding may add stuff to pages I think.
But I'll let @enj comment on that

Reply:
Yeah customer branding would definitely be an issue here, you would have to make the CSP configurable. Seems outside the scope of this particular PR.

Reply:
Indeed @jwforres , I am going to make an Issue and we can resolve that in 3.8
CSP sounds like something we may want to share with Web console as far as using a common option when customer needs changes, perhaps.